client/src/Components/Login.jsx

```diff
@@ -73,10 +65,14 @@ function Login() {
     }
   };
 
-  useEffect(() => {
-    // eslint-disable-next-line no-unused-expressions
-    data.email ? login({ password: data.password, email: data.email }, call) : null;
-  }, [data]);
+  const handelSubmit = (e) => {
+    e.preventDefault();
+    setData({
+      email,
+      password,
+    });
```

Comment:
this state is unnecessary to you can pass the email and password without setting it